### DISCUSSION TITLE
Debugging the project system

### DISCUSSION THREAD

**Discussion by RobertvanderHulst** [dotnet/project-system#9530] [Q&A]
I managed to successfully debug the project system and run it from inside VS. 
I can set breakpoints everywhere and debug most of the areas.
Now I want to debug into the project property pages.
I would expect that the CSharpProjectDesignerPageProvider class would be used to build the list of property pages, but the GetPages() method never gets called.
I also have set break points in the VB code for the various property pages, but the debugger also does not stop there.
When I look in the Debugger-Modules window, then I can see that the symbols are loaded.
The debugger shows normal breakpoint icons in the margin, but does not stop at the breakpoints.
I tried the debugger with and without 'Just my code' and I have also disabled the "require source files to exactly match..." option.
What could be wrong?

Background: I am creating a project system for our own managed language, based on the code in the C#/VB/F# project system.

**Comment by RobertvanderHulst** [comment]
To answer my own question: If you supply the CPS project system with information through PropertyPageSchema items, then most of the code inside the VB projects will be never called.

**Reply by drewnoakes** [reply]
If a project has the `ProjectPropertiesEditor` project capability defined, then the legacy (VB) property pages are not used.
